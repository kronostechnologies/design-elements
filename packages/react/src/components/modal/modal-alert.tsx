import React, { ReactElement } from 'react';
import { AbstractProps, ModalAbstract } from './modal-abstract';

export function ModalAlert({ ...props }: AbstractProps): ReactElement {
    return <ModalAbstract modalType="alert" {...props}/>;
}
