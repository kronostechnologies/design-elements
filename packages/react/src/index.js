export Card from './components/card';
export Enso from './components/enso';
export Headband from './components/headband';
export Progress from './components/progress';
export RollingEnso from './components/enso';
export SVG from './components/svg';

export PrimaryButton from './components/buttons/primary';
export SecondaryButton from './components/buttons/secondary';

export PrimaryAddButton from './components/buttons/add/primary';
export SecondaryAddButton from './components/buttons/add/secondary';

export Checkbox from './components/inputs/checkbox';
export OptionButton from './components/inputs/option';
export SearchInput from './components/inputs/search';
export Select from './components/inputs/select';
export TextInput from './components/inputs/text';
