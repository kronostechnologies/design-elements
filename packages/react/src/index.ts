// Buttons
export { Button } from './components/buttons/button';
export { IconButton } from './components/buttons/icon-button';
export { NavListOption } from './components/nav-list/nav-list-option';
export { MenuButton } from './components/menu-button/menu-button';
export { DropdownNavigation } from './components/dropdown-navigation/dropdown-navigation';
export { BentoMenuButton } from './components/bento-menu-button/bento-menu-button';
export { ToggleButtonGroup } from './components/toggle-button-group/toggle-button-group';

// Form Elements
export { Avatar } from './components/avatar/avatar';
export { Checkbox } from './components/checkbox/checkbox';
export { CheckboxGroup } from './components/checkbox-group/checkbox-group';
export { Datepicker, DatepickerHandles } from './components/date-picker/date-picker';
export { DropdownList } from './components/dropdown-list/dropdown-list';
export { MoneyInput } from './components/money-input/money-input';
export { OptionButton } from './components/option-button/option-button';
export { PasswordCreationInput } from './components/password-creation-input/password-creation-input';
export { PhoneInput } from './components/phone-input/phone-input';
export { RadioButtonGroup } from './components/radio-button-group/radio-button-group';
export { SearchContextual } from './components/search/search-contextual';
export { SearchGlobal } from './components/search/search-global';
export { StepperInput } from './components/stepper-input/stepper-input';
export { Tab, Tabs } from './components/tabs/tabs';
export { TextArea } from './components/text-area/text-area';
export { TextInput } from './components/text-input/text-input';
export { ToggleSwitch } from './components/toggle-switch/toggle-switch';
export {
    hasALowerCaseLetter,
    hasAnUpperCaseLetter,
    hasMinLength as isLongEnough,
} from './components/password-creation-input/validation-condition';

// Context
export { ThemeWrapper } from './components/theme-wrapper/theme-wrapper';
export { DeviceContextProvider, useDeviceContext } from './components/device-context-provider/device-context-provider';
export { IntlProvider } from './components/internationalization-provider/internationalization-provider';

// Lists
export { Listbox, ListboxOption } from './components/listbox/listbox';

// Miscellaneous
export { Badge } from './components/badge/badge';
export { Card } from './components/card/card';
export { CardLink } from './components/card-link/card-link';
export * from './components/carousel/carousel';
export { ChooserButtonGroup, ChooserButtonOption } from './components/chooser-button-group/chooser-button-group';
export { ErrorSummary, ErrorMessage } from './components/error-summary/error-summary';
export { ExternalLink } from './components/external-link/external-link';
export { Heading } from './components/heading/heading';
export { GlobalHeader } from './components/global-header/global-header';
export { DropdownMenuButton } from './components/dropdown-menu-button/dropdown-menu-button';
export { ExternalItemProps } from './components/dropdown-menu/list-items/external-item';
export { NavItemProps } from './components/dropdown-menu/list-items/nav-item';
export { Icon } from './components/icon/icon';
export { SectionalBanner, SectionalBannerType } from './components/sectional-banner/sectional-banner';
export * from './components/progress-tracker/progress-tracker';
export { SideDrawer } from './components/side-drawer/side-drawer';
export { GlobalBanner, GlobalBannerType } from './components/global-banner/global-banner';
export * from './components/tooltip/tooltip';
export * from './components/toggletip/toggletip';
export * from './components/table/table';
export { Modal } from './components/modal/modal';
export { ModalDialog } from './components/modal/modal-dialog';
export { Tag, TagProps, TagValue } from './components/tag/tag';
export { Lozenge } from './components/lozenge/lozenge';
export { RadioCard } from './components/radio-card-group/radio-card';
export { RadioCardGroup } from './components/radio-card-group/radio-card-group';
export { Status, StatusType } from './components/status/status';
export { SkipLink } from './components/skip-link/skip-link';
export { Spinner } from './components/spinner/spinner';
export { useModal } from './components/modal/use-modal';
export { UserProfile } from './components/user-profile/user-profile';
export { ToastType } from './components/toast/toast-type';
export { useToast } from './hooks/use-toast';

// Navigation
export { Breadcrumb, BreadcrumbElement } from './components/breadcrumb/breadcrumb';
export * from './components/global-navigation/global-navigation';
export { RouteLink } from './components/route-link/route-link';

// Results
export * from './components/legend/legend';
export { ProgressIndicator } from './components/progress-indicator/progress-indicator';
export { ProgressCircle } from './components/progress-circle/progress-circle';
export { Pagination } from './components/pagination/pagination';

// Themes
export { equisoftTheme, customTheme } from './themes';
export { injectMainCss } from './styles';
export { Theme } from './themes/theme';

// Hooks
export { useTheme } from './hooks/use-theme';

export * from './components/design-system';
