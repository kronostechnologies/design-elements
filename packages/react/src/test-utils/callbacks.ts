export function doNothing(): void {
    // do nothing
}
