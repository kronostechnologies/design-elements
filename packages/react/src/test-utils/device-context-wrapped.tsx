import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
