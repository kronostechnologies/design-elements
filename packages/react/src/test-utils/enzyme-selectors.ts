import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
