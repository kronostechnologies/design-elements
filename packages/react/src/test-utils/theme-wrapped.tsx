import React, { ComponentProps, ComponentType, ReactElement } from 'react';
import { equisoftTheme, ThemeWrapper } from '..';

type ThemeProps = Pick<ComponentProps<typeof ThemeWrapper>, 'children'>;

export function themeProvider(): ComponentType {
    return  ThemeProvider as ComponentType;
}

export function ThemeProvider({ children }: ThemeProps): ReactElement {
    return (
        <ThemeWrapper theme={equisoftTheme}>
            {children}
        </ThemeWrapper>
    );
}

export function ThemeWrapped(children: ReactElement): ReactElement {
    return (
        <ThemeWrapper theme={equisoftTheme}>
            {children}
        </ThemeWrapper>
    );
}
