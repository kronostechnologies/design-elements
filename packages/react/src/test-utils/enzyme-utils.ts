import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';

export async function waitForComponentToPaint<P = {}>(
    wrapper: ReactWrapper<P>,
    amount = 0,
): Promise<void> {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, amount));
        wrapper.update();
    });
}

export function expectFocusToBeOn(element: ReactWrapper): void {
    expect(document.activeElement).toBe(element.getDOMNode());
}
