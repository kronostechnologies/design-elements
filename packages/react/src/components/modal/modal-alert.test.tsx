import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { render } from '@testing-library/react';
import React from 'react';
import { ModalAlert } from './modal-alert';
jest.mock('uuid/v4');

describe('Modal Alert', () => {
    test('Matches snapshot (opened)', () => {
        const tree = render(
            ThemeWrapped(
                <ModalAlert
                    ariaHideApp={false}
                    title="Title"
                    subtitle="Subtitle"
                    isOpen={true}
                    onRequestClose={() => {}}
                >
                    Test Content
                </ModalAlert>,
            ),
        );

        expect(tree).toMatchSnapshot();
    });

    test('Matches snapshot (not titles)', () => {
        const tree = render(
            ThemeWrapped(
                <ModalAlert
                    ariaHideApp={false}
                    isOpen={true}
                    onRequestClose={() => {}}
                >
                    Test Content
                </ModalAlert>,
            ),
        );

        expect(tree).toMatchSnapshot();
    });

    test('Matches snapshot (closed)', () => {
        const tree = render(
            ThemeWrapped(
                <ModalAlert
                    ariaHideApp={false}
                    title="Title"
                    subtitle="Subtitle"
                    isOpen={false}
                    onRequestClose={() => {}}
                >
                    Test Content
                </ModalAlert>,
            ),
        );

        expect(tree).toMatchSnapshot();
    });
});
