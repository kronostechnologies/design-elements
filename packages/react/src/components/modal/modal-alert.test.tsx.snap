// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Alert Matches snapshot (closed) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class=""
  >
    <div />
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Alert Matches snapshot (not titles) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div />
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.6rem;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.55px;
  -moz-letter-spacing: 0.55px;
  -ms-letter-spacing: 0.55px;
  letter-spacing: 0.55px;
  min-height: 2rem;
  min-width: 2rem;
  outline: none;
  padding: var(--spacing-half) var(--spacing-2x);
  text-transform: uppercase;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #EF483E;
  border-color: #EF483E;
  color: #FFFFFF;
}

.c2:not(:disabled) {
  cursor: pointer;
}

.c2 > svg {
  color: inherit;
}

.c2:hover {
  background-color: #D41F14;
  border-color: #D41F14;
}

.c2:disabled,
.c2:disabled:focus,
.c2:disabled:hover {
  background-color: #F9B6B2;
  border-color: #F9B6B2;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.6rem;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.55px;
  -moz-letter-spacing: 0.55px;
  -ms-letter-spacing: 0.55px;
  letter-spacing: 0.55px;
  min-height: 2rem;
  min-width: 2rem;
  outline: none;
  padding: var(--spacing-half) var(--spacing-2x);
  text-transform: uppercase;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border-color: transparent;
  color: #57666E;
}

.c4:not(:disabled) {
  cursor: pointer;
}

.c4 > svg {
  color: inherit;
}

.c4:hover {
  color: #000000;
}

.c4:disabled,
.c4:disabled:focus,
.c4:disabled:hover {
  color: #9CA7B4;
}

.c0 {
  background-color: #FFFFFF;
  border: 1px solid #57666E;
  border-radius: var(--border-radius);
  box-shadow: 0 6px 10px 0 rgba(0,0,0,0.1);
  box-sizing: border-box;
  padding: var(--spacing-4x);
  position: relative;
  width: 500px;
}

.c1 {
  margin-top: var(--spacing-4x);
}

.c3 {
  margin-right: var(--spacing-1x);
}

<div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay ReactModal__Overlay--after-open"
        style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75); align-items: center; display: flex; justify-content: center;"
      >
        <div
          aria-modal="true"
          class="ReactModal__Content ReactModal__Content--after-open c0"
          role="alertdialog"
          tabindex="-1"
        >
          Test Content
          <footer
            class="c1"
          >
            <button
              class="c2 c3"
              label="Confirm"
              type="submit"
            >
              Confirm
            </button>
            <button
              class="c4 "
              label="Cancel"
              type="submit"
            >
              Cancel
            </button>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Alert Matches snapshot (opened) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.6rem;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.55px;
  -moz-letter-spacing: 0.55px;
  -ms-letter-spacing: 0.55px;
  letter-spacing: 0.55px;
  min-height: 2rem;
  min-width: 2rem;
  outline: none;
  padding: var(--spacing-half) var(--spacing-2x);
  text-transform: uppercase;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #EF483E;
  border-color: #EF483E;
  color: #FFFFFF;
}

.c5:not(:disabled) {
  cursor: pointer;
}

.c5 > svg {
  color: inherit;
}

.c5:hover {
  background-color: #D41F14;
  border-color: #D41F14;
}

.c5:disabled,
.c5:disabled:focus,
.c5:disabled:hover {
  background-color: #F9B6B2;
  border-color: #F9B6B2;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.6rem;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.55px;
  -moz-letter-spacing: 0.55px;
  -ms-letter-spacing: 0.55px;
  letter-spacing: 0.55px;
  min-height: 2rem;
  min-width: 2rem;
  outline: none;
  padding: var(--spacing-half) var(--spacing-2x);
  text-transform: uppercase;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border-color: transparent;
  color: #57666E;
}

.c7:not(:disabled) {
  cursor: pointer;
}

.c7 > svg {
  color: inherit;
}

.c7:hover {
  color: #000000;
}

.c7:disabled,
.c7:disabled:focus,
.c7:disabled:hover {
  color: #9CA7B4;
}

.c0 {
  background-color: #FFFFFF;
  border: 1px solid #57666E;
  border-radius: var(--border-radius);
  box-shadow: 0 6px 10px 0 rgba(0,0,0,0.1);
  box-sizing: border-box;
  padding: var(--spacing-4x);
  position: relative;
  width: 500px;
}

.c1 {
  margin-bottom: var(--spacing-3x);
}

.c4 {
  margin-top: var(--spacing-4x);
}

.c2 {
  font-size: 1.25rem;
  font-weight: var(--font-normal);
  height: 32px;
  line-height: 24px;
  margin-bottom: var(--spacing-2x);
  margin-top: 0;
}

.c3 {
  font-size: 1rem;
  font-weight: var(--font-normal);
  height: 24px;
  line-height: 22px;
  margin: 0;
}

.c6 {
  margin-right: var(--spacing-1x);
}

<body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div />
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay ReactModal__Overlay--after-open"
        style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75); align-items: center; display: flex; justify-content: center;"
      >
        <div
          aria-label="Title"
          aria-modal="true"
          class="ReactModal__Content ReactModal__Content--after-open c0"
          role="alertdialog"
          tabindex="-1"
        >
          <header
            class="c1"
          >
            <h2
              class="c2"
              tabindex="-1"
            >
              Title
            </h2>
            <h3
              class="c3"
              tabindex="-1"
            >
              Subtitle
            </h3>
          </header>
          Test Content
          <footer
            class="c4"
          >
            <button
              class="c5 c6"
              label="Confirm"
              type="submit"
            >
              Confirm
            </button>
            <button
              class="c7 "
              label="Cancel"
              type="submit"
            >
              Cancel
            </button>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
